s3/s3.go

```diff
@@ -201,14 +206,14 @@ func (b *Bucket) GetReader(path string) (rc io.ReadCloser, err error) {
 
 // PutReader inserts an object into the S3 bucket by consuming data
 // from r until EOF.
-func (b *Bucket) PutReader(path string, r io.Reader, length int64, contType string, perm ACL) error {
+func (b *Bucket) PutReader(path string, r io.ReadSeeker, length int64, contType string, perm ACL) error {
```

Comment:
+1 also the name doesn't match if it's a ReadSeeker. How about adding separate method?